src/uct/tcp/sockcm/sockcm_md.c

```diff
@@ -43,10 +43,105 @@ ucs_status_t uct_sockcm_md_query(uct_md_h md, uct_md_attr_t *md_attr)
     return UCS_OK;
 }
 
+static int uct_sockcm_is_addr_route_resolved(int sock_id, struct sockaddr *addr,
+                                            int addrlen)
+{
+    char ip_port_str[UCS_SOCKADDR_STRING_LEN];
+    char host[UCS_SOCKADDR_STRING_LEN];
+    char serv[UCS_SOCKADDR_STRING_LEN];
+    int ret = -1;
+
+    ret = getnameinfo(addr, addrlen, host, UCS_SOCKADDR_STRING_LEN, serv,
+                      UCS_SOCKADDR_STRING_LEN, NI_NAMEREQD);
+    if (0 != ret) {
+        ucs_debug("getnameinfo error : %s\n", gai_strerror(ret));
+        return 0;
+    }
+
+    if (connect(sock_id, addr, addrlen)) {
+
+        if (errno == ECONNREFUSED) {
+            return 1;
+        }
+
+        ucs_debug("connect(addr = %s) failed: %m",
+                   ucs_sockaddr_str(addr, ip_port_str, UCS_SOCKADDR_STRING_LEN));
+        return 0;
+    }
+
+    return 1;
+}
+
+static int uct_sockcm_is_sockaddr_inaddr_any(struct sockaddr *addr)
```

Comment:
can you use uct_rdmacm_is_sockaddr_inaddr_any() ? otherwise it's code duplication.
i mean, create a common function both rdmacm and sockcm would use.

Reply:
done